### PR TITLE
[ZEPPELIN-3725] Fix SQL injection

### PR DESCRIPTION
**Pull request by vtkhanh**
### What type of PR is it?
Security - SQL Injection

### Todos
* [ ] - Task

### What is the Jira issue?
* [ZEPPELIN-3725](https://issues.apache.org/jira/browse/ZEPPELIN-3725)

### How should this be tested?
* No behavior change
* Existing tests passed.

### Questions:
* Does the license files need to update? NO
* Is there breaking changes for older versions? NO
* Does this needs documentation? NO

### CODE REVIEW

**Review by jongyoul — DISMISSED** [review]
Thank you for your contribution. LGTM

File: zeppelin-server/src/main/java/org/apache/zeppelin/service/ShiroAuthenticationService.java [hunk 1]
```diff
@@ -434,7 +434,7 @@ private List<String> getUserList(JdbcRealm obj) {
         return userlist;
       }
 
-      userquery = String.format("SELECT %s FROM %s", username, tablename);
+      userquery = String.format("SELECT ? FROM ?", username, tablename);
```
Comment (Reamer) [inline zeppelin-server/src/main/java/org/apache/zeppelin/service/ShiroAuthenticationService.java:437]: I think that `String.format` is no longer necessary. What do you think?
```suggestion
      userquery = "SELECT ? FROM ?";
```

**Review by Reamer — CHANGES_REQUESTED** [review]
Thank you for your pull request. I have only one small question.

**Review by pan3793 — APPROVED** [review]
@vtkhanh thanks for fixing, may I ask is there any known exploit scenario or just a potential security issue?

**Review by iemre — APPROVED** [review]

File: zeppelin-server/src/main/java/org/apache/zeppelin/service/ShiroAuthenticationService.java (hunk 1 above)
Comment (vtkhanh) [reply zeppelin-server/src/main/java/org/apache/zeppelin/service/ShiroAuthenticationService.java]: +1. Thanks. Updated.

**Comment by vtkhanh** [comment]
> @vtkhanh thanks for fixing, may I ask is there any known exploit scenario or just a potential security issue?

No, there is not an exploit that I'm aware of. But there is prolonged [CVE](https://security.snyk.io/vuln/SNYK-JAVA-ORGAPACHEZEPPELIN-73629) reported that we came across, and the fix should be straightforward, so I go for it.

**Comment by pan3793** [comment]
@vtkhanh thanks for clarification.

**Review by Reamer — APPROVED** [review]
LGTM

**Comment by vtkhanh** [comment]
This [CI](https://ci.appveyor.com/project/ApacheSoftwareFoundation/zeppelin/builds/48294783) run is timeout, anyone knows if this is a common issue for this CI? 
Could I somehow rerun the CI?

**Comment by iemre** [comment]
> This [CI](https://ci.appveyor.com/project/ApacheSoftwareFoundation/zeppelin/builds/48294783) run is timeout, anyone knows if this is a common issue for this CI? Could I somehow rerun the CI?

I normally run below to re-trigger CI.
```
git commit —amend —no-edit
git push -f
```

**Comment by vtkhanh** [comment]
> > This [CI](https://ci.appveyor.com/project/ApacheSoftwareFoundation/zeppelin/builds/48294783) run is timeout, anyone knows if this is a common issue for this CI? Could I somehow rerun the CI?
> 
> I normally run below to re-trigger CI.
> 
> ```
> git commit —amend —no-edit
> git push -f
> ```

Thanks Emre. It does help.

**Comment by vtkhanh** [comment]
The CI passed. Could you help to merge the PR once you get a chance, @jongyoul ? Thank you.

**Comment by jongyoul** [comment]
@vtkhanh Hello, I ran CI. It's the first contribution so some committers need to run CI manually. Could you please just wait until the CI finishes?

**Pull request merged**